app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,48 @@
+class ApplicationsController < ApplicationController
+  def show 
+    @application = Application.find(params[:id])
+    if params[:pet_search]
+      @searched_pets = Pet.search(params[:pet_search])
+    end
+  end
+
+  def new
+
+  end
+
+  def create
+    application = Application.create(application_params)
+    if application.save
+      redirect_to "/applications/#{application.id}"
+    else
+      flash[:alert] = "Please fill in all fields."
+      render :new
+    end
+  end
+
+  def update
+    application = Application.find(params[:id])
+    pets = application.pets
+    application.update(
+      adoption_description: params[:adoption_description],
+      status: "Pending"
+    )
+    application.save
+    application.pets = pets
+
+    redirect_to "/applications/#{application.id}"
+  end
+
+  def add_pet
```

Comment:
A more conventional/RESTful option here would be to put this functionality in a `create` action in the PetApplicationsController

Also, see note in application model but you can add a pet in a controller rather than encapsulating it at the model level. That's appropriate CRUD to leave in the controller!
In general, these AR methods are totally fine in a controller:
`.create`
`.save`
`.new`
`.update`
`.destroy`
`.find`
`.find_by`

It's also totally fine to call a model's associations in the controller, i.e.
`@application.pets`